### ISSUE TITLE
Allow customization of Access Token Scope

### ISSUE THREAD

**Issue by MartinFlores751** [irods/irods_client_http_api#248]
We should allow the user to specify additional scopes along with the default `oidc` scope for further flexibility.

Example (Implementation may be different):
```json
"oidc": {
  "additional_scopes": ["irods", "custom", "misc"]
}
```

**Comment by MartinFlores751** [comment]
Perhaps even allow the `oidc` scope to be optional?

**Comment by korydraughn** [comment]
We haven't spent any time thinking through the details due to other priorities, but we'll address this once work on 0.4.0 begins.

**Comment by korydraughn** [comment]
> By validating scopes (a scope mapping could be added in the conf to establish a relation between scope names and API ops), the resource server will expose only those API endpoints compatible with them, increasing security and providing a way of restricting API access dynamically.

True. iRODS already has a permission model though. Admins can already block access to endpoints (e.g. /data-objects) through the use of a proxy (e.g. nginx or httpd). While we can certainly improve the use of scopes, it's difficult to determine whether it'll prove to be useful to people.

That's just my initial thoughts.

We haven't made any decisions regarding this yet. We're very much open to hearing use-cases from people. Feel free to post your thoughts and ideas here.

**Comment by korydraughn** [comment]
Thanks. This is good feedback. We'll keep this in mind as we approach the next release.

If you think of anything else, please post it here.
